src/tictactoe_clojure/computer_player.clj

```diff
@@ -4,46 +4,77 @@
   (:require [tictactoe-clojure.board :as board])
   (:require [tictactoe-clojure.rules :as rules]))
 
-(def max-cell-num-computable 12)
 (def preferred-positions-on-4x4 #{1 4 13 16})
-(def preferred-positions-on-5x5 #{13 1 5 21 25})
-
-(defn- create-pair
-  [coll-1 coll-2]
-  (partition 2 (interleave coll-1 coll-2)))
-
-(declare optimal-move-by-score)
-(declare score-board)
-
-(defn- score-board
-  [player board depth is-max-player]
-  (let [winner (rules/winner board)
-        other-player (rules/switch-turn player)]
-    (cond
-      (= (rules/draw? board) true) 0
-      (= winner player)
-      (if is-max-player
-        (- 10 depth)
-        (+ -10 depth))
-      (= winner other-player)
-      (if is-max-player
-        (+ -10 depth)
-        (- 10 depth))
-      :else
-        (second
-          (optimal-move-by-score other-player board (dec depth) (not is-max-player))))))
-
-(def memo-score-board (memoize score-board))
-
-(defn- optimal-move-by-score
-  [player board depth is-max-player]
-  (let [possible-moves (board/valid-slots board)
-        score-of-possible-move (map #(memo-score-board player (board/move board % player) depth is-max-player) possible-moves)
-        move-score-pair (create-pair possible-moves score-of-possible-move)
-        sorted-move-scores-pair (sort-by second move-score-pair)]
+(def preferred-positions-on-5x5 #{1 3 5 11 13 21 25})
+(def initial-alpha -100)
+(def initial-beta 100)
+
+(defn- assign-initial-score
+  [is-max-player]
+    (if is-max-player
+      [0 -100]
+      [0 100]))
+
+(defn- winning-score
+  [board depth max-player]
+    (if (= max-player (rules/winner board))
+      (+ 10 depth)
+      (- -10 depth)))
+
+(defn- gameover-move-score
+  [board depth max-player]
+    (if (rules/winner? board)
+      [0 (winning-score board depth max-player)]
+      [0 0]))
+
+(defn- find-max
+  [max-player best-position valued-position move-position]
+    (if (< (second best-position) (second valued-position))
+      (vector move-position (second valued-position))
+      best-position))
+
+(defn- find-min
```

Comment:
`find-max` and `find-min` are by and large the same with the exception of the comparator. Can you think of a way to DRY this up?